lib/rubocop/cop/style/redundant_regexp_argument.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Identifies places where argument can be replaced from
+      # a deterministic regexp to a string.
+      #
+      # @example
+      #   # bad
+      #   'foo'.byteindex(/f/)
+      #   'foo'.byterindex(/f/)
+      #   'foo'.gsub(/f/, 'x')
+      #   'foo'.gsub!(/f/, 'x')
+      #   'foo'.index(/f/)
+      #   'foo'.partition(/f/)
+      #   'foo'.rindex(/f/)
+      #   'foo'.rpartition(/f/)
+      #   'foo'.scan(/f/)
+      #   'foo'.split(/f/)
+      #   'foo'.sub(/f/, 'x')
+      #   'foo'.sub!(/f/, 'x')
+      #
+      #   # good
+      #   'foo'.byteindex('f')
+      #   'foo'.byterindex('f')
+      #   'foo'.gsub('f', 'x')
+      #   'foo'.gsub!('f', 'x')
+      #   'foo'.index('f')
+      #   'foo'.partition('f')
+      #   'foo'.rindex('f')
+      #   'foo'.rpartition('f')
+      #   'foo'.scan('f')
+      #   'foo'.split('f')
+      #   'foo'.sub('f', 'x')
+      #   'foo'.sub!('f', 'x')
+      class RedundantRegexpArgument < Base
+        extend AutoCorrector
+
+        MSG = 'Use string `%<prefer>s` as argument instead of regexp `%<current>s`.'
+        RESTRICT_ON_SEND = %i[
+          byteindex byterindex gsub gsub! index partition rindex rpartition scan split sub sub!
```

Comment:
```suggestion
          [] []= byteindex byterindex gsub gsub! index partition rindex rpartition scan slice slice! split start_with? sub sub!
```

Reply:
Good catch! However, `[]` and `[]=` are not applicable because they are false positives when the receiver is a hash. Others have applied :-)

Reply:
could we check the receiver? `index`, `rindex`, `partition`, `slice`, `slice!` are also Array/Enumerable/Hash methods and could lead to false positives otherwise.

Reply:
Ah, it's true that `slice` and `slice!` are bad when the receiver is a `Hash` object, so l've removed them. Other methods can usually be kept since an Integer is used as an argument when the receiver is an `Array` and `Enumerable`, I think.

Reply:
hmm, thats true for `partition`, `slice` and `slice!`, which only take a regexp when called on strings, but not for `index` and `rindex`:

```ruby
[/a/, /b/, /b/, /c/].index /b/  # => 1
[/a/, /b/, /b/, /c/].rindex /b/ # => 2
```